### PR TITLE
Refactor exception handling in StatusReplyTest

### PR DESCRIPTION
**Pull request by Captain1653**
It's my first and simple PR to akka. 
I have noticed some lines `try .. catch` in tests StatusReplyTest. 
I think these lines can be replaced on [assertThrows from junit 4.13](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.String,%20java.lang.Class,%20org.junit.function.ThrowingRunnable))

I'm not sure about separated issue for the PR. I can create the issue if it's necessary.  Any advice will be appreciated.

Kind regards

### CODE REVIEW

File: akka-actor-tests/src/test/java/akka/pattern/StatusReplyTest.java [hunk 1]
```diff
@@ -30,53 +30,41 @@
...
     assertFalse(reply.isError());
     assertEquals("woho", reply.getValue());
-    try {
-      reply.getError();
-      Assert.fail("Calling get error on success did not throw");
-    } catch (IllegalArgumentException ex) {
-      // this is what we expect
-    }
+    Assert.assertThrows(
+        "Calling get error on success did not throw",
```
Comment (raboof) [inline akka-actor-tests/src/test/java/akka/pattern/StatusReplyTest.java:39]: When using 'assertXxx' notation, I personally prefer using 'should' in the description like this:

```suggestion
        "Calling .getError() on success should throw",
```

I think this makes reading the test code easier, and still gives a useful hint when the test fails.

**Review by raboof — COMMENTED** [review]
Thanks for taking the time to prepare this PR!

The change in general makes sense - especially for the case with multiple 'catch' blocks it looks nicer.

I think it would be nicer to write the descriptions of the assert statements in 'should' form.

**Review by octonato — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM, nice, thanks for contributing

**Pull request merged**